File: training-data/objectivec/WordFrequency.m
#import <Foundation/Foundation.h>

NSDictionary *wordFrequency(NSString *text) {
    NSMutableDictionary *freq = [NSMutableDictionary dictionary];
    NSArray *words = [text.lowercaseString componentsSeparatedByString:@" "];
    for (NSString *word in words) {
        NSNumber *count = freq[word];
        freq[word] = @(count.integerValue + 1);
    }
    return freq;
}

int main(int argc, const char *argv[]) {
    @autoreleasepool {
        NSLog(@"%@", wordFrequency(@"the quick brown fox the lazy fox"));
    }
    return 0;
}

File: training-data/prolog/word_frequency.pl
word_frequency(Text, Freq) :-
    string_lower(Text, Lower),
    split_string(Lower, " ", "", Words),
    msort(Words, Sorted),
    count_words(Sorted, Freq).

count_words([], []).
count_words([W|Ws], [W-Count|Rest]) :-
    partition([X]>>(X == W), [W|Ws], Same, Different),
    length(Same, Count),
    count_words(Different, Rest).

:- word_frequency("the quick brown fox the lazy fox", Freq), writeln(Freq).
